li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -532,13 +581,17 @@ public void close() {
     close(CONSUMER_CLOSE_MAX_TIMEOUT_SECS, TimeUnit.SECONDS);
   }
 
-  void closeNoLock(long timeout, TimeUnit timeUnit) {
-    if (_closed) {
-      return;
-    }
+  @Override
+  synchronized public void close(Duration timeout) {
+    close(timeout.toMillis(), TimeUnit.MILLISECONDS);
+  }
 
-    _closed = true;
+  // This is used for testing only
+  public FederatedSubscriptionState getCurrentSubscription() {
+    return _currentSubscription;
+  }
 
+  void closeNoLock(long timeout, TimeUnit timeUnit) {
```

Comment:
private? (or explain why package protected)

Reply:
Added private and updated to use Duration.